R/extract_long_term_disease.R

```diff
@@ -0,0 +1,168 @@
+#' Extraction des Affections Longue Durée (ALD)
+#'
+#' Cette fonction permet d'extraire des ALD actives au
+#' moins un jour sur une période donnée.
+#' Les ALD dont l'intersection [IMB_ALD_DTD, IMB_ALD_DTF]
+#' avec la période [start_date, end_date] n'est pas vide
+#' sont extraites.
+#' Si des codes ICD 10 ou des numéros d'ALD sont fournis,
+#' seules les ALD associées à ces codes ICD 10 ou numéros
+#' d'ALD sont extraites.
+#' Si des identifiants de patients sont fournis, seules
+#' les ALD associées à ces patients sont extraites.
+#'
+#' @param start_date Date La date de début de la période
+#' sur laquelle extraire les ALD actives.
+#' @param end_date Date La date de fin de la période
+#' sur laquelle extraire les ALD actives.
+#' @param icd_cod_starts_with character vector Un vecteur de codes
+#' ICD 10. Si `icd_cod_starts_with` ou `ald_numbers` sont fournis,
+#' seules les ALD associées à ces codes ICD 10 ou numéros d'ALD
+#' sont extraites. Sinon, toutes les ALD actives sur la période
+#' [start_date, end_date] sont extraites.
+#' @param ald_numbers numeric vector Un vecteur de numéros d'ALD.
+#' Si `icd_cod_starts_with` ou `ald_numbers` sont fournis,
+#' seules les ALD associées à ces codes ICD 10 ou numéros d'ALD
+#' sont extraites. Sinon, toutes les ALD actives sur la période
+#' [start_date, end_date] sont extraites.
+#' @param excl_atm_nat character vector Un vecteur de codes
+#' IMB_ATM_NAT à exclure. Par défaut, les ALD de nature
+#' 11, 12 et 13 sont exclues.
+#' @param patients_ids data.frame Un data.frame contenant les
+#' paires d'identifiants des patients pour lesquels les ALD
+#' doivent être extraites. Les colonnes de ce data.frame
+#' doivent être "ben_idt_ano" et "ben_nir_psa" (en minuscules). Les
+#' "ben_nir_psa" doivent être tous les "ben_nir_psa" associés aux
+#' "ben_idt_ano" fournis. Si `patients_ids` n'est pas fourni,
+#' les ALD de tous les patients sont extraites.
+#' @return Un data.frame contenant les ALDs actives sur la période.
+#' La première colonne est BEN_NIR_PSA si aucun identifiant de
+#' patient n'est fourni. Si  la première colonne est BEN_IDT_ANO.
+#' Les autres colonnes par défaut sont les suivantes :
+#' - IMB_ALD_NUM : Le numéro de l'ALD
+#' - IMB_ALD_DTD : La date de début de l'ALD
+#' - IMB_ALD_DTF : La date de fin de l'ALD
+#' - IMB_ETM_NAT : La nature de l'ALD
+#' - MED_MTF_COD : Le code ICD 10 de la pathologie associée à l'ALD
+#'
+#' @examples
+#' start_date <- as.Date("2010-01-01")
```

Comment:
Il faut inclure l'exemple dans un  \dontrun{} comme [documenté ici](https://stackoverflow.com/questions/12038160/how-to-not-run-an-example-using-roxygen2#12038225). 

Dans le cas contraire, la construction des pages de documentation R va planter car il essayera de lancer ce code et va se planter car il va essayer d'initier une connection à la base oracle.

Reply:
Merci pour l'info, je modifie ça !